crates/primitives/src/snapshot/segment.rs

```diff
@@ -81,17 +93,26 @@ impl SnapshotSegment {
     }
 
     /// Takes a filename and parses the [`SnapshotSegment`] and its inclusive range.
-    pub fn parse_filename(name: &str) -> Option<(Self, RangeInclusive<BlockNumber>)> {
+    pub fn parse_filename(
+        name: &str,
```

Comment:
should we have a few helper functions that deal with Osstr etc, so we don't necessarily need to alloc a string when we parse a file name?

Reply:
it takes osstr now